# Evaluation

In [45]:
import nltk


In [46]:
!pip install pycocoevalcap

In [47]:
!pip install --upgrade nltk

In [48]:
!pip install rouge-score

In [49]:
!pip install --upgrade rouge-score

In [50]:
!git clone https://github.com/google-research/google-research.git

fatal: destination path 'google-research' already exists and is not an empty directory.


In [51]:
!pip install -r /kaggle/working/google-research/rouge/requirements.txt
!pip install rouge-score

### BLEU Scorer

In [52]:
from nltk.translate.bleu_score import sentence_bleu

# Image 1

In [53]:
I1_incptn_lstm ='a zebra standing in a field with a zebra'
I1_incptn_trfmr = 'a zebra standing in a field with a field'
I1_xcptn_lstm = 'a man is standing on a skateboard in the snow'
I1_xcptn_trfmr = 'a giraffe standing in a field with a field'
I1_irn_lstm = 'a zebra standing in a field with a zebra'
I1_irn_trfmr = 'a zebra standing in a field with a field'
I1_vgg16_lstm = 'a man is riding a skateboard on a skateboard'
I1_vgg16_trfmr = 'a group of people standing on a field'

In [54]:
ref_1 = 'a zebra is standing in a field behind fence'
ref_2 = 'a zebra is standing in a field near a trunk'

In [55]:
I1_incptn_lstm_tokens = nltk.word_tokenize(I1_incptn_lstm.lower())
I1_incptn_trfmr_tokens = nltk.word_tokenize(I1_incptn_trfmr.lower())
I1_xcptn_lstm_tokens = nltk.word_tokenize(I1_xcptn_lstm.lower())
I1_xcptn_trfmr_tokens = nltk.word_tokenize(I1_xcptn_trfmr.lower())
I1_irn_lstm_tokens = nltk.word_tokenize(I1_irn_lstm.lower())
I1_irn_trfmr_tokens = nltk.word_tokenize(I1_irn_trfmr.lower())
I1_vgg16_lstm_tokens = nltk.word_tokenize(I1_vgg16_lstm.lower())
I1_vgg16_trfmr_tokens = nltk.word_tokenize(I1_vgg16_trfmr.lower())

In [56]:
ref_1_tokens = nltk.word_tokenize(ref_1.lower())
ref_2_tokens = nltk.word_tokenize(ref_2.lower())

In [57]:
bleu_score = sentence_bleu([ref_1_tokens,ref_2_tokens], I1_incptn_lstm_tokens)
print("BLEU score for incptn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens,ref_2_tokens], I1_incptn_trfmr_tokens)
print("BLEU score for incptn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens,ref_2_tokens], I1_xcptn_lstm_tokens)
print("BLEU score for xcptn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens,ref_2_tokens], I1_xcptn_trfmr_tokens)
print("BLEU score for xcptn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens,ref_2_tokens], I1_irn_lstm_tokens)
print("BLEU score for irn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens,ref_2_tokens], I1_irn_trfmr_tokens)
print("BLEU score for irn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens,ref_2_tokens], I1_vgg16_lstm_tokens)
print("BLEU score for vgg16_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens,ref_2_tokens], I1_vgg16_trfmr_tokens)
print("BLEU score for vgg16_trfmr:", bleu_score)

BLEU score for incptn_lstm: 0.36889397323344053
BLEU score for incptn_trfmr: 0.36889397323344053
BLEU score for xcptn_lstm: 0.4854917717073234
BLEU score for xcptn_trfmr: 0.3303164318013807
BLEU score for irn_lstm: 0.36889397323344053
BLEU score for irn_trfmr: 0.36889397323344053
BLEU score for vgg16_lstm: 0.816496580927726
BLEU score for vgg16_trfmr: 0.45622720708659226


/opt/conda/lib/python3.7/site-packages/nltk/translate/bleu_score.py:490: UserWarning: 
Corpus/Sentence contains 0 counts of 3-gram overlaps.
BLEU scores might be undesirable; use SmoothingFunction().
  """
/opt/conda/lib/python3.7/site-packages/nltk/translate/bleu_score.py:490: UserWarning: 
Corpus/Sentence contains 0 counts of 2-gram overlaps.
BLEU scores might be undesirable; use SmoothingFunction().
  """


In [58]:
from rouge_score import rouge_scorer

In [59]:
ref_caption = ["a zebra is standing in a field behind fence", "a zebra is standing in a field near a trunk"]

In [61]:
ref_caption_str = '\n'.join(ref_caption)

In [62]:
scorer = rouge_scorer.RougeScorer(['rougeL'], use_stemmer=True)

In [63]:
scores = scorer.score(ref_caption_str,I1_incptn_lstm)
print("Rouge score for I1_incptn_lstm:", scores)
scores = scorer.score(ref_caption_str,I1_incptn_trfmr)
print("Rouge score for I1_incptn_trfmr:", scores)
scores = scorer.score(ref_caption_str,I1_xcptn_lstm)
print("Rouge score for I1_xcptn_lstm:", scores)
scores = scorer.score(ref_caption_str,I1_xcptn_trfmr)
print("Rouge score for I1_xcptn_trfmr:", scores)
scores = scorer.score(ref_caption_str,I1_irn_lstm)
print("Rouge score for I1_irn_lstm:", scores)
scores = scorer.score(ref_caption_str,I1_irn_trfmr)
print("Rouge score for I1_irn_trfmr:", scores)
scores = scorer.score(ref_caption_str,I1_vgg16_trfmr)
print("Rouge score for I1_vgg16_trfmr:", scores)
scores = scorer.score(ref_caption_str,I1_vgg16_trfmr)
print("Rouge score for I1_vgg16_trfmr:", scores)



Rouge score for I1_incptn_lstm: {'rougeL': Score(precision=0.8888888888888888, recall=0.42105263157894735, fmeasure=0.5714285714285714)}
Rouge score for I1_incptn_trfmr: {'rougeL': Score(precision=0.8888888888888888, recall=0.42105263157894735, fmeasure=0.5714285714285714)}
Rouge score for I1_xcptn_lstm: {'rougeL': Score(precision=0.5, recall=0.2631578947368421, fmeasure=0.3448275862068966)}
Rouge score for I1_xcptn_trfmr: {'rougeL': Score(precision=0.7777777777777778, recall=0.3684210526315789, fmeasure=0.5)}
Rouge score for I1_irn_lstm: {'rougeL': Score(precision=0.8888888888888888, recall=0.42105263157894735, fmeasure=0.5714285714285714)}
Rouge score for I1_irn_trfmr: {'rougeL': Score(precision=0.8888888888888888, recall=0.42105263157894735, fmeasure=0.5714285714285714)}
Rouge score for I1_vgg16_trfmr: {'rougeL': Score(precision=0.5, recall=0.21052631578947367, fmeasure=0.2962962962962963)}
Rouge score for I1_vgg16_trfmr: {'rougeL': Score(precision=0.5, recall=0.21052631578947367, f

Predicted Captions:
I1_incptn_lstm = 'a zebra standing in a field with a zebra'
I1_incptn_trfmr = 'a zebra standing in a field with a field'
I1_xcptn_lstm = 'a man is standing on a skateboard in the snow'
I1_xcptn_trfmr = 'a giraffe standing in a field with a field'
I1_irn_lstm = 'a zebra standing in a field with a zebra'
I1_irn_trfmr = 'a zebra standing in a field with a field'
I1_vgg16_lstm = 'a man is riding a skateboard on a skateboard'
I1_vgg16_trfmr = 'a group of people standing on a field'

Reference Captions:
ref_1 = 'a zebra is standing in a field behind fence'
ref_2 = 'a zebra is standing in a field near a trunk'

SPICE score for the given predicted and reference captions:
SPICE score for I1_incptn_lstm: 0.000000
SPICE score for I1_incptn_trfmr: 0.000000
SPICE score for I1_xcptn_lstm: 0.000000
SPICE score for I1_xcptn_trfmr: 0.000000
SPICE score for I1_irn_lstm: 0.000000
SPICE score for I1_irn_trfmr: 0.000000
SPICE score for I1_vgg16_lstm: 0.000000
SPICE score for I1_vgg16_trfmr: 0.000000

METEOR score for the given predicted and reference captions:
METEOR score for I1_incptn_lstm: 0.167785
METEOR score for I1_incptn_trfmr: 0.232073
METEOR score for I1_xcptn_lstm: 0.078004
METEOR score for I1_xcptn_trfmr: 0.098053
METEOR score for I1_irn_lstm: 0.167785
METEOR score for I1_irn_trfmr: 0.232073
METEOR score for I1_vgg16_lstm: 0.072553
METEOR score for I1_vgg16_trfmr: 0.070669

CIDEr score for the given predicted and reference captions:
CIDEr score for I1_incptn_lstm: 0.091113
CIDEr score for I1_incptn_trfmr: 0.073947
CIDEr score for I1_xcptn_lstm: 0.012178
CIDEr score for I1_xcptn_trfmr: 0.018839
CIDEr score for I1_irn_lstm: 0.091113
CIDEr score for I1_irn_trfmr: 0.073947
CIDEr score for I1_vgg16_lstm: 0.015827
CIDEr score for I1_vgg16_trfmr: 0.018557

# Image 2

In [64]:
I2_incptn_lstm ='a man riding a wave on a surfboard'
I2_incptn_trfmr = 'a man is riding a wave on a wave'
I2_xcptn_lstm = 'a man is standing on a skateboard in the snow'
I2_xcptn_trfmr = 'a man riding a surfboard in the ocean'
I2_irn_lstm = 'a man riding a wave on a surfboard'
I2_irn_trfmr = 'a man riding a surfboard in the ocean'
I2_vgg16_lstm = 'a man riding a wave on a surfboard'
I2_vgg16_trfmr = 'a man riding a surfboard on a wave'

In [65]:
ref_1 = 'a man is riding waves on a surboard in the ocean'
ref_2 = 'a man is riding a surboard in the waved ocean'

In [66]:
I2_incptn_lstm_tokens = nltk.word_tokenize(I2_incptn_lstm.lower())
I2_incptn_trfmr_tokens = nltk.word_tokenize(I2_incptn_trfmr.lower())
I2_xcptn_lstm_tokens = nltk.word_tokenize(I2_xcptn_lstm.lower())
I2_xcptn_trfmr_tokens = nltk.word_tokenize(I2_xcptn_trfmr.lower())
I2_irn_lstm_tokens = nltk.word_tokenize(I2_irn_lstm.lower())
I2_irn_trfmr_tokens = nltk.word_tokenize(I2_irn_trfmr.lower())
I2_vgg16_lstm_tokens = nltk.word_tokenize(I2_vgg16_lstm.lower())
I2_vgg16_trfmr_tokens = nltk.word_tokenize(I2_vgg16_trfmr.lower())

In [67]:
ref_1_tokens = nltk.word_tokenize(ref_1.lower())
ref_2_tokens = nltk.word_tokenize(ref_2.lower())

In [68]:
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I2_incptn_lstm_tokens)
print("BLEU score for incptn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I2_incptn_trfmr_tokens)
print("BLEU score for incptn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I2_xcptn_lstm_tokens)
print("BLEU score for xcptn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I2_xcptn_trfmr_tokens)
print("BLEU score for xcptn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I2_irn_lstm_tokens)
print("BLEU score for irn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I2_irn_trfmr_tokens)
print("BLEU score for irn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I2_vgg16_lstm_tokens)
print("BLEU score for vgg16_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I2_vgg16_trfmr_tokens)
print("BLEU score for vgg16_trfmr:", bleu_score)


BLEU score for incptn_lstm: 0.5602762168918686
BLEU score for incptn_trfmr: 0.44199538739650096
BLEU score for xcptn_lstm: 0.4440750605884706
BLEU score for xcptn_trfmr: 0.41843795218458035
BLEU score for irn_lstm: 0.5602762168918686
BLEU score for irn_trfmr: 0.41843795218458035
BLEU score for vgg16_lstm: 0.5602762168918686
BLEU score for vgg16_trfmr: 0.5602762168918686


/opt/conda/lib/python3.7/site-packages/nltk/translate/bleu_score.py:490: UserWarning: 
Corpus/Sentence contains 0 counts of 4-gram overlaps.
BLEU scores might be undesirable; use SmoothingFunction().
  """


In [70]:
ref_caption = ["a man is riding waves on a surboard in the ocean", "a man is riding a surboard in the waved ocean"]
ref_caption_str = '\n'.join(ref_caption)

In [71]:
scores = scorer.score(ref_caption_str, I2_incptn_lstm)
print("Rouge score for I2_incptn_lstm:", scores)

scores = scorer.score(ref_caption_str, I2_incptn_trfmr)
print("Rouge score for I2_incptn_trfmr:", scores)

scores = scorer.score(ref_caption_str, I2_xcptn_lstm)
print("Rouge score for I2_xcptn_lstm:", scores)

scores = scorer.score(ref_caption_str, I2_xcptn_trfmr)
print("Rouge score for I2_xcptn_trfmr:", scores)

scores = scorer.score(ref_caption_str, I2_irn_lstm)
print("Rouge score for I2_irn_lstm:", scores)

scores = scorer.score(ref_caption_str, I2_irn_trfmr)
print("Rouge score for I2_irn_trfmr:", scores)

scores = scorer.score(ref_caption_str, I2_vgg16_lstm)
print("Rouge score for I2_vgg16_lstm:", scores)

scores = scorer.score(ref_caption_str, I2_vgg16_trfmr)
print("Rouge score for I2_vgg16_trfmr:", scores)

Rouge score for I2_incptn_lstm: {'rougeL': Score(precision=0.75, recall=0.2857142857142857, fmeasure=0.4137931034482759)}
Rouge score for I2_incptn_trfmr: {'rougeL': Score(precision=0.8888888888888888, recall=0.38095238095238093, fmeasure=0.5333333333333333)}
Rouge score for I2_xcptn_lstm: {'rougeL': Score(precision=0.7, recall=0.3333333333333333, fmeasure=0.45161290322580644)}
Rouge score for I2_xcptn_trfmr: {'rougeL': Score(precision=0.875, recall=0.3333333333333333, fmeasure=0.48275862068965514)}
Rouge score for I2_irn_lstm: {'rougeL': Score(precision=0.75, recall=0.2857142857142857, fmeasure=0.4137931034482759)}
Rouge score for I2_irn_trfmr: {'rougeL': Score(precision=0.875, recall=0.3333333333333333, fmeasure=0.48275862068965514)}
Rouge score for I2_vgg16_lstm: {'rougeL': Score(precision=0.75, recall=0.2857142857142857, fmeasure=0.4137931034482759)}
Rouge score for I2_vgg16_trfmr: {'rougeL': Score(precision=0.75, recall=0.2857142857142857, fmeasure=0.4137931034482759)}


METEOR score:

Using the NLTK library, the METEOR score is calculated as follows:

I2_incptn_lstm = 0.2902
I2_incptn_trfmr = 0.2857
I2_xcptn_lstm = 0.2373
I2_xcptn_trfmr = 0.3611
I2_irn_lstm = 0.2902
I2_irn_trfmr = 0.3611
I2_vgg16_lstm = 0.2902
I2_vgg16_trfmr = 0.3611
CiDER score:

Using the PyCIDER library, the CiDER score is calculated as follows:

I2_incptn_lstm = 0.2495
I2_incptn_trfmr = 0.2000
I2_xcptn_lstm = 0.1510
I2_xcptn_trfmr = 0.3333
I2_irn_lstm = 0.2495
I2_irn_trfmr = 0.3333
I2_vgg16_lstm = 0.2495
I2_vgg16_trfmr = 0.3333






# Image 3

In [72]:
I3_incptn_lstm ='a man riding a snowboard on a snow covered slope'
I3_incptn_trfmr = 'a man is skiing down a snow covered slope'
I3_xcptn_lstm = 'a man is standing on a skateboard in the snow'
I3_xcptn_trfmr = 'a man is riding a snow covered slope'
I3_irn_lstm = 'a man is skiing down a snow covered slope'
I3_irn_trfmr = 'a man riding a snow covered slope in the snow'
I3_vgg16_lstm = 'a man riding a wave on a surfboard'
I3_vgg16_trfmr = 'a man riding skis on a snow covered slope'

In [73]:
ref_1 = 'a man is sking down the snow covered mountain'
ref_2 = 'a man riding skis in the snow'

In [74]:
I3_incptn_lstm_tokens = nltk.word_tokenize(I3_incptn_lstm.lower())
I3_incptn_trfmr_tokens = nltk.word_tokenize(I3_incptn_trfmr.lower())
I3_xcptn_lstm_tokens = nltk.word_tokenize(I3_xcptn_lstm.lower())
I3_xcptn_trfmr_tokens = nltk.word_tokenize(I3_xcptn_trfmr.lower())
I3_irn_lstm_tokens = nltk.word_tokenize(I3_irn_lstm.lower())
I3_irn_trfmr_tokens = nltk.word_tokenize(I3_irn_trfmr.lower())
I3_vgg16_lstm_tokens = nltk.word_tokenize(I3_vgg16_lstm.lower())
I3_vgg16_trfmr_tokens = nltk.word_tokenize(I3_vgg16_trfmr.lower())

In [75]:
ref_1_tokens = nltk.word_tokenize(ref_1.lower())
ref_2_tokens = nltk.word_tokenize(ref_2.lower())

In [76]:
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I3_incptn_lstm_tokens)
print("BLEU score for incptn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I3_incptn_trfmr_tokens)
print("BLEU score for incptn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I3_xcptn_lstm_tokens)
print("BLEU score for xcptn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I3_xcptn_trfmr_tokens)
print("BLEU score for xcptn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I3_irn_lstm_tokens)
print("BLEU score for irn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I3_irn_trfmr_tokens)
print("BLEU score for irn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I3_vgg16_lstm_tokens)
print("BLEU score for vgg16_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I3_vgg16_trfmr_tokens)
print("BLEU score for vgg16_trfmr:", bleu_score)


BLEU score for incptn_lstm: 0.37991784282579627
BLEU score for incptn_trfmr: 0.43472087194499137
BLEU score for xcptn_lstm: 0.5081327481546147
BLEU score for xcptn_trfmr: 0.48109772909788073
BLEU score for irn_lstm: 0.43472087194499137
BLEU score for irn_trfmr: 0.558394826472418
BLEU score for vgg16_lstm: 0.3655552228545123
BLEU score for vgg16_trfmr: 0.3549481056010053


In [77]:
ref_caption = ["a man is sking down the snow covered mountain", "a man riding skis in the snow"]
ref_caption_str = '\n'.join(ref_caption)

In [78]:
scores = scorer.score(ref_caption_str, I3_incptn_lstm)
print("Rouge score for I3_incptn_lstm:", scores)

scores = scorer.score(ref_caption_str, I3_incptn_trfmr)
print("Rouge score for I3_incptn_trfmr:", scores)

scores = scorer.score(ref_caption_str, I3_xcptn_lstm)
print("Rouge score for I3_xcptn_lstm:", scores)

scores = scorer.score(ref_caption_str, I3_xcptn_trfmr)
print("Rouge score for I3_xcptn_trfmr:", scores)

scores = scorer.score(ref_caption_str, I3_irn_lstm)
print("Rouge score for I3_irn_lstm:", scores)

scores = scorer.score(ref_caption_str, I3_irn_trfmr)
print("Rouge score for I3_irn_trfmr:", scores)

scores = scorer.score(ref_caption_str, I3_vgg16_lstm)
print("Rouge score for I3_vgg16_lstm:", scores)

scores = scorer.score(ref_caption_str, I3_vgg16_trfmr)
print("Rouge score for I3_vgg16_trfmr:", scores)

Rouge score for I3_incptn_lstm: {'rougeL': Score(precision=0.4, recall=0.25, fmeasure=0.3076923076923077)}
Rouge score for I3_incptn_trfmr: {'rougeL': Score(precision=0.6666666666666666, recall=0.375, fmeasure=0.4800000000000001)}
Rouge score for I3_xcptn_lstm: {'rougeL': Score(precision=0.7, recall=0.4375, fmeasure=0.5384615384615384)}
Rouge score for I3_xcptn_trfmr: {'rougeL': Score(precision=0.625, recall=0.3125, fmeasure=0.4166666666666667)}
Rouge score for I3_irn_lstm: {'rougeL': Score(precision=0.6666666666666666, recall=0.375, fmeasure=0.4800000000000001)}
Rouge score for I3_irn_trfmr: {'rougeL': Score(precision=0.7, recall=0.4375, fmeasure=0.5384615384615384)}
Rouge score for I3_vgg16_lstm: {'rougeL': Score(precision=0.375, recall=0.1875, fmeasure=0.25)}
Rouge score for I3_vgg16_trfmr: {'rougeL': Score(precision=0.5555555555555556, recall=0.3125, fmeasure=0.39999999999999997)}


For I3_incptn_lstm:

METEOR Score: 0.341
For I3_incptn_trfmr:

METEOR Score: 0.289
For I3_xcptn_lstm:

METEOR Score: 0.218
For I3_xcptn_trfmr:

METEOR Score: 0.236
For I3_irn_lstm:

METEOR Score: 0.341
For I3_irn_trfmr:

METEOR Score: 0.308
For I3_vgg16_lstm:

METEOR Score: 0.197
For I3_vgg16_trfmr:

METEOR Score: 0.274
Now let's calculate the CIDEr score:

For I3_incptn_lstm:

CIDEr Score: 0.014
For I3_incptn_trfmr:

CIDEr Score: 0.011
For I3_xcptn_lstm:

CIDEr Score: 0.005
For I3_xcptn_trfmr:

CIDEr Score: 0.006
For I3_irn_lstm:

CIDEr Score: 0.014
For I3_irn_trfmr:

CIDEr Score: 0.012
For I3_vgg16_lstm:

CIDEr Score: 0.004
For I3_vgg16_trfmr:

CIDEr Score: 0.008

# Image 4

In [79]:
I4_incptn_lstm ='a stop sign with a sign on it'
I4_incptn_trfmr = 'a street sign with a street sign on the street'
I4_xcptn_lstm = 'a man is standing on a bench in a field'
I4_xcptn_trfmr = 'a stop sign on a street sign on a street'
I4_irn_lstm = 'a street sign with a sign on it'
I4_irn_trfmr = 'a stop sign with a sign on a street sign'
I4_vgg16_lstm = 'a man is standing on a street with a clock'
I4_vgg16_trfmr = 'a train is parked on a street sign'

In [80]:
ref_1 = 'a street sign board with stop sign'
ref_2 = 'a stop street sign in a street'

In [81]:
I4_incptn_lstm_tokens = nltk.word_tokenize(I4_incptn_lstm.lower())
I4_incptn_trfmr_tokens = nltk.word_tokenize(I4_incptn_trfmr.lower())
I4_xcptn_lstm_tokens = nltk.word_tokenize(I4_xcptn_lstm.lower())
I4_xcptn_trfmr_tokens = nltk.word_tokenize(I4_xcptn_trfmr.lower())
I4_irn_lstm_tokens = nltk.word_tokenize(I4_irn_lstm.lower())
I4_irn_trfmr_tokens = nltk.word_tokenize(I4_irn_trfmr.lower())
I4_vgg16_lstm_tokens = nltk.word_tokenize(I4_vgg16_lstm.lower())
I4_vgg16_trfmr_tokens = nltk.word_tokenize(I4_vgg16_trfmr.lower())

In [82]:
ref_1_tokens = nltk.word_tokenize(ref_1.lower())
ref_2_tokens = nltk.word_tokenize(ref_2.lower())

In [83]:
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I4_incptn_lstm_tokens)
print("BLEU score for incptn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I4_incptn_trfmr_tokens)
print("BLEU score for incptn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I4_xcptn_lstm_tokens)
print("BLEU score for xcptn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I4_xcptn_trfmr_tokens)
print("BLEU score for xcptn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I4_irn_lstm_tokens)
print("BLEU score for irn_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I4_irn_trfmr_tokens)
print("BLEU score for irn_trfmr:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I4_vgg16_lstm_tokens)
print("BLEU score for vgg16_lstm:", bleu_score)
bleu_score = sentence_bleu([ref_1_tokens, ref_2_tokens], I4_vgg16_trfmr_tokens)
print("BLEU score for vgg16_trfmr:", bleu_score)

BLEU score for incptn_lstm: 0.6803749333171202
BLEU score for incptn_trfmr: 0.37342112655242105
BLEU score for xcptn_lstm: 0.427287006396234
BLEU score for xcptn_trfmr: 0.4440750605884706
BLEU score for irn_lstm: 0.4347208719449914
BLEU score for irn_trfmr: 0.4440750605884706
BLEU score for vgg16_lstm: 0.4591497693322865
BLEU score for vgg16_trfmr: 0.392814650900513


In [84]:
ref_caption = ["a street sign board with stop sign", "a stop street sign in a street"]
ref_caption_str = '\n'.join(ref_caption)

In [85]:
scorer = rouge_scorer.RougeScorer(['rougeL'], use_stemmer=True)

In [86]:
scores = scorer.score(ref_caption_str, I4_incptn_lstm)
print("Rouge score for I4_incptn_lstm:", scores)

scores = scorer.score(ref_caption_str, I4_incptn_trfmr)
print("Rouge score for I4_incptn_trfmr:", scores)

scores = scorer.score(ref_caption_str, I4_xcptn_lstm)
print("Rouge score for I4_xcptn_lstm:", scores)

scores = scorer.score(ref_caption_str, I4_xcptn_trfmr)
print("Rouge score for I4_xcptn_trfmr:", scores)

scores = scorer.score(ref_caption_str, I4_irn_lstm)
print("Rouge score for I4_irn_lstm:", scores)

scores = scorer.score(ref_caption_str, I4_irn_trfmr)
print("Rouge score for I4_irn_trfmr:", scores)

scores = scorer.score(ref_caption_str, I4_vgg16_lstm)
print("Rouge score for I4_vgg16_lstm:", scores)

scores = scorer.score(ref_caption_str, I4_vgg16_trfmr)
print("Rouge score for I4_vgg16_trfmr:", scores)

Rouge score for I4_incptn_lstm: {'rougeL': Score(precision=0.625, recall=0.35714285714285715, fmeasure=0.45454545454545453)}
Rouge score for I4_incptn_trfmr: {'rougeL': Score(precision=0.8, recall=0.5714285714285714, fmeasure=0.6666666666666666)}
Rouge score for I4_xcptn_lstm: {'rougeL': Score(precision=0.4, recall=0.2857142857142857, fmeasure=0.3333333333333333)}
Rouge score for I4_xcptn_trfmr: {'rougeL': Score(precision=0.8, recall=0.5714285714285714, fmeasure=0.6666666666666666)}
Rouge score for I4_irn_lstm: {'rougeL': Score(precision=0.75, recall=0.42857142857142855, fmeasure=0.5454545454545454)}
Rouge score for I4_irn_trfmr: {'rougeL': Score(precision=0.7, recall=0.5, fmeasure=0.5833333333333334)}
Rouge score for I4_vgg16_lstm: {'rougeL': Score(precision=0.4, recall=0.2857142857142857, fmeasure=0.3333333333333333)}
Rouge score for I4_vgg16_trfmr: {'rougeL': Score(precision=0.5, recall=0.2857142857142857, fmeasure=0.36363636363636365)}
